webui/src/lib/components/requireAuth.tsx

```diff
@@ -0,0 +1,18 @@
+import React from "react";
+import {Navigate, Outlet, useLocation} from "react-router-dom";
+import {useAuth} from "../auth/authContext";
+import {AUTH_STATUS} from "../auth/status";
+
+export default function RequireAuth() {
```

Comment:
Nit but important - should be `RequiresAuth`, since it's a trait of a component.